cirq/contrib/paulistring/pauli_string_raw_types.py

```diff
@@ -63,10 +63,7 @@ def _pauli_string_diagram_info(self,
         qubits = self.qubits if args.known_qubits is None else args.known_qubits
         syms = tuple('[{}]'.format(self.pauli_string[qubit])
                      for qubit in qubits)
-        if exponent_absorbs_sign and self.pauli_string.negated:
-            if isinstance(exponent, float):
-                exponent = -exponent
-            else:
-                exponent = '-{!s}'.format(exponent)
+        if exponent_absorbs_sign and self.pauli_string.coefficient == -1:
```

Comment:
This looks suspicious. What if coefficient is nearly -1, e.g. `exp(1j * pi * 99/100)`?

At minimum, we should raise. Better, find a way to handle all phases.

Reply:
Added a TODO.

This refactoring is somewhat large and I want to tackle it in pieces.